### PR TITLE
VIH-10652 add state management

### PR DESCRIPTION
**Pull request by shaed-parkar**
### Jira link (if applicable)

VIH-10652

### Change description ###

* add ngrx
* create new models for state management
* replace status list and status controls to use the new store

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/angular.json [hunk 1]
```diff
@@ -1,192 +1,178 @@
...
-                },
-                "lint": {
-                    "builder": "@angular-eslint/builder:lint",
-                    "options": {
-                        "lintFilePatterns": [
-                            "src/**/*.ts",
-                            "src/**/*.html"
-                        ]
-                    }
+  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
```
Comment (oliver-scott) [inline VideoWeb/VideoWeb/ClientApp/angular.json:2]: are there any changes needed to this file? It's a bit tricky to tell with the changes to the formatting

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts [hunk 2]
```diff
@@ -47,7 +42,7 @@ export class JudgeParticipantStatusListComponent extends WRParticipantStatusList
     }
 
     ngOnInit() {
-        this.hearing = new Hearing(this.conference);
+        this.hearing = new Hearing(this.conference as any); // TODO: create a new Hearing ctor that accepts VHConference
```
Comment (oliver-scott) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-participant-status-list/judge-participant-status-list.component.ts:45]: there's a TODO here - for a future PR?

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts [hunk 3]
```diff
@@ -241,6 +249,21 @@ export abstract class WaitingRoomBaseDirective implements AfterContentChecked {
...
+        this.store
+            .select(ConferenceSelectors.getActiveConference)
+            .pipe(takeUntil(this.onDestroy$))
+            .subscribe(conf => {
+                this.vhConference = conf;
+                // this does not need to be an observable and we can move away from the flag service
+                this.phoneNumber$ = this.vhConference.isVenueScottish
+                    ? of(this.contactDetails.scotland.phoneNumber)
+                    : of(this.contactDetails.englandAndWales.phoneNumber);
+                // this.hearingVenueFlagsService.hearingVenueIsScottish$.pipe(
```
Comment (oliver-scott) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts:261]: some commented out code here

File: azure-pipelines.sds.pr-release.yml [hunk 4]
```diff
@@ -1,185 +1,187 @@
...
 
 resources:
-    repositories:
-        - repository: azTemplates
-          type: github
-          name: hmcts/azure-devops-templates
-          ref: master
-          endpoint: hmcts
+  repositories:
+    - repository: azTemplates
```
Comment (oliver-scott) [inline azure-pipelines.sds.pr-release.yml:5]: as per a previous comment, are there any changes to make to this file? Looks like you might need to change the tab settings in your editor

**Review by oliver-scott — COMMENTED** [review]
looks good, just a few comments

**Review by oliver-scott — APPROVED** [review]

**Review by komaldabhi1509 — APPROVED** [review]

**Review by himesh99 — APPROVED** [review]
lgtm

**Pull request merged**